test-core/src/main/java/com/adyen/checkout/test/extensions/FlowExtension.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by onurk on 31/1/2023.
+ */
+
+package com.adyen.checkout.test.extensions
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+
+@OptIn(ExperimentalCoroutinesApi::class)
+fun <T> TestScope.testFlow(
```

Comment:
I think we should change the name of this method or change the way it works, it's not obvious what it does now

Reply:
What do you think `collectFlowEventsIntoListForTest`?

Reply:
I'm not sure we should keep it like this, the problem I have with its implementation is that it takes a mutable list as an argument but modifies its contents which is really not obvious.